### PR TITLE
Issue 103 upstream arg vals

### PR DESCRIPTION
**Pull request by DavidGoldwasser**
This enhances the method to get argument values from upstream measures to cleanly handle non-string argument types so measures no longer have to hard code in logic for this. Tests were added to a new spec file for helper_methods

### CODE REVIEW

**Comment by DavidGoldwasser** [comment]
Addresses issue #103

File: lib/openstudio/extension/core/os_lib_helper_methods.rb [hunk 1]
```diff
@@ -265,7 +265,20 @@ def self.check_upstream_measure_for_arg(runner, arg_name)
           result = measure_step.result.get
           result.stepValues.each do |arg|
             name = arg.name
-            value = arg.valueAsVariant.to_s
+            # check if value, double, int, or bool
+            value_type = arg.variantType.valueDescription
```
Comment (nllong) [inline lib/openstudio/extension/core/os_lib_helper_methods.rb:269]: is this an openstudio change?

File: spec/openstudio/core/os_lib_geometry_spec.rb [hunk 2]
```diff
@@ -55,4 +55,4 @@
       expect(res.length).to eq 96
     end
   end
-end
+end
```
Comment (nllong) [inline spec/openstudio/core/os_lib_geometry_spec.rb:58]:
```suggestion
end

```

**Review by nllong — COMMENTED** [review]
fantastic! Couple comments.

File: lib/openstudio/extension/core/os_lib_helper_methods.rb (hunk 1 above)
Comment (DavidGoldwasser) [reply lib/openstudio/extension/core/os_lib_helper_methods.rb:269]: There was no change in OpenStudio, I'm just making this more robust so it doesn't return everything as a string, which is what was happening before. Then in the measure I had added code to convert back to integer or double in specific use cases. Now that should just work without any special code in the measure.

**Pull request merged**